// @sushiswap/trident/types/factories/IBalancerV1__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type { IBalancerV1, IBalancerV1Interface } from "../IBalancerV1";

const _abi = [
  {
    inputs: [
      {
        internalType: "uint256",
        name: "poolAmountIn",
        type: "uint256",
      },
      {
        internalType: "uint256[]",
        name: "minAmountsOut",
        type: "uint256[]",
      },
    ],
    name: "exitPool",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [],
    name: "getFinalTokens",
    outputs: [
      {
        internalType: "address[]",
        name: "",
        type: "address[]",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
];

export class IBalancerV1__factory {
  static readonly abi = _abi;
  static createInterface(): IBalancerV1Interface {
    return new utils.Interface(_abi) as IBalancerV1Interface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): IBalancerV1 {
    return new Contract(address, _abi, signerOrProvider) as IBalancerV1;
  }
}
