// @sushiswap/trident/types/factories/TridentBatchable__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  TridentBatchable,
  TridentBatchableInterface,
} from "../TridentBatchable";

const _abi = [
  {
    inputs: [
      {
        internalType: "bytes[]",
        name: "data",
        type: "bytes[]",
      },
    ],
    name: "batch",
    outputs: [
      {
        internalType: "bytes[]",
        name: "results",
        type: "bytes[]",
      },
    ],
    stateMutability: "payable",
    type: "function",
  },
];

export class TridentBatchable__factory {
  static readonly abi = _abi;
  static createInterface(): TridentBatchableInterface {
    return new utils.Interface(_abi) as TridentBatchableInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): TridentBatchable {
    return new Contract(address, _abi, signerOrProvider) as TridentBatchable;
  }
}
